pycyphal/transport/udp/_ip/_endpoint_mapping.py

```diff
@@ -60,23 +72,26 @@ def node_id_to_unicast_ip(local_ip_address: IPAddress, node_id: int) -> IPAddres
       ...
     ValueError: Invalid node-ID...
     """
-    if node_id > IP_ADDRESS_NODE_ID_MASK:
-        raise ValueError(f"Invalid node-ID: {node_id} is larger than {IP_ADDRESS_NODE_ID_MASK}")
+    if data_specifier.service_id > SERVICE_ID_MASK:
+        raise ValueError(f"Invalid node-ID: {data_specifier.subject_id} is larger than {SERVICE_ID_MASK}")
```

Comment:
Subject-IDs, service-IDs, and node-IDs are all distinct entities. Subject- and service-IDs are sometimes referred to by their generalized name of "port-ID".